nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,143 @@
+                        + str(per_page_query_parameter)
+                        + "&"
+                        + "page="
+                        + str(page_query_parameter)
+                    )
+            else:
+                # TODO handle the case where there is no instances of the model is found
+                self.std.out.write(f"No instances of {model_class} found")
+        else:
+            # A generic endpoint like `core:home`
+            if view_name not in self.app_name_to_urls["endpoints"]:
+                self.app_name_to_urls["endpoints"][view_name] = []
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+
+    def fetch_urls(self, url_patterns):
+        """
+        Store the URL patterns in the dictionary to output an .YAML file
+        The dictionary will have the following structure:
+        {
+            "endpoints": {
+                <app_name>:<view_name>: [
+                    <url_pattern_1>,
+                    <url_pattern_2>,
+                ],
+                dcim:device: [
+                    "/dcim/devices/cfbd447f-d563-4fac-bb75-bdda70ab4e80/",
+                    "/dcim/devices/38471bfe-0aca-4e09-b545-b0f90280fb66/",
+                ],
+                ...
+            },
+            ...
+        """
+        for pattern in url_patterns:
+            if hasattr(pattern, "url_patterns"):
+                # If it's a nested URL pattern, recursively list its URLs
+                self.fetch_urls(pattern.url_patterns)
+            else:
+                # TODO need to include Nautobot App endpoints as well at some point
+                if pattern.lookup_str.startswith(
+                    (
+                        "nautobot.circuits.views",
+                        "nautobot.cloud.views",
+                        "nautobot.core.views",
+                        "nautobot.dcim.views",
+                        "nautobot.extras.views",
+                        "nautobot.ipam.views",
+                        "nautobot.tenancy.views",
+                        "nautobot.virtualization.views",
+                        "nautobot.wireless.views",
+                    )
+                ):
+                    # Only fetch urls from relevant apps
+                    app_name = pattern.lookup_str.split(".")[1]
+                    model = pattern.default_args.get("model", None)
+                    if model:
+                        app_name = model._meta.app_label
+
+                    view_name = f"{app_name}:{pattern.name}"
+                    url_pattern = f"/{app_name}/{pattern.pattern}"
+
+                    if self.is_get_endpoint(view_name):
+                        # Replace "^" and "$" from the url pattern
+                        url_pattern = url_pattern.replace("^", "").replace("$", "")
+                        # Retrieve the model class for the view name
+                        model_class = get_model_for_view_name(view_name)
```

Comment:
This might be redundant with line 131 above but `model` could be `None` for models with `NautobotUIViewSet-based` views. So this is necessary.